### ISSUE TITLE
KeyProvider18.java line 90

### ISSUE THREAD

**Issue by owdeveloper** [aws-amplify/aws-sdk-android#1108]
**Describe the bug**
We're getting a very high volume of AWS library crashes in Firebase Crashlytics, could you check?

**Which AWS service(s) are affected?**
Android SDK 2.13.5
`implementation "com.amazonaws:aws-android-sdk-core:2.13.5"`
`implementation "com.amazonaws:aws-android-sdk-cognitoidentityprovider:2.13.5"`
`implementation "com.amazonaws:aws-android-sdk-appsync:2.10.0"`

**Screenshots**
![image](https://user-images.githubusercontent.com/6125289/62375159-020eab80-b514-11e9-9241-1d230a4193e7.png)
![image](https://user-images.githubusercontent.com/6125289/62375271-48fca100-b514-11e9-9d59-5edf1de10492.png)

**Additional context**
Firebase Crashlytics.
```
Fatal Exception: java.lang.RuntimeException: Unable to create application com.pontomobi.smiles.SmilesApplication: java.lang.IllegalStateException: Error in initializing the CognitoCachingCredentialsProvider. 
       at android.app.ActivityThread.handleBindApplication + 4592(ActivityThread.java:4592)
       at android.app.ActivityThread.access$1500 + 155(ActivityThread.java:155)
       at android.app.ActivityThread$H.handleMessage + 1303(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 136(Looper.java:136)
       at android.app.ActivityThread.main + 5426(ActivityThread.java:5426)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke + 515(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1268(ZygoteInit.java:1268)
       at com.android.internal.os.ZygoteInit.main + 1084(ZygoteInit.java:1084)
       at dalvik.system.NativeStart.main(NativeStart.java)
```
```
Caused by java.lang.IllegalStateException: Error in initializing the CognitoCachingCredentialsProvider. 
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 434(CognitoCachingCredentialsProvider.java:434)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1014(Instrumentation.java:1014)
       at android.app.ActivityThread.handleBindApplication + 4589(ActivityThread.java:4589)
       at android.app.ActivityThread.access$1500 + 155(ActivityThread.java:155)
       at android.app.ActivityThread$H.handleMessage + 1303(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 136(Looper.java:136)
       at android.app.ActivityThread.main + 5426(ActivityThread.java:5426)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke + 515(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1268(ZygoteInit.java:1268)
       at com.android.internal.os.ZygoteInit.main + 1084(ZygoteInit.java:1084)
       at dalvik.system.NativeStart.main(NativeStart.java)
```
```
Caused by java.lang.IllegalStateException: java.lang.IllegalArgumentException: key == null
       at com.amazonaws.internal.keyvaluestore.KeyProvider18.getKey + 108(KeyProvider18.java:108)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.setPersistenceEnabled + 138(AWSKeyValueStore.java:138)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.<init> + 108(AWSKeyValueStore.java:108)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 425(CognitoCachingCredentialsProvider.java:425)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1014(Instrumentation.java:1014)
       at android.app.ActivityThread.handleBindApplication + 4589(ActivityThread.java:4589)
       at android.app.ActivityThread.access$1500 + 155(ActivityThread.java:155)
       at android.app.ActivityThread$H.handleMessage + 1303(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 136(Looper.java:136)
       at android.app.ActivityThread.main + 5426(ActivityThread.java:5426)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke + 515(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1268(ZygoteInit.java:1268)
       at com.android.internal.os.ZygoteInit.main + 1084(ZygoteInit.java:1084)
       at dalvik.system.NativeStart.main(NativeStart.java)
```
```
Caused by java.lang.IllegalArgumentException: key == null
       at javax.crypto.spec.SecretKeySpec.<init> + 59(SecretKeySpec.java:59)
       at com.amazonaws.internal.keyvaluestore.KeyProvider18.getKey + 90(KeyProvider18.java:90)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.setPersistenceEnabled + 138(AWSKeyValueStore.java:138)
       at com.amazonaws.internal.keyvaluestore.AWSKeyValueStore.<init> + 108(AWSKeyValueStore.java:108)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.initialize + 425(CognitoCachingCredentialsProvider.java:425)
       at com.amazonaws.auth.CognitoCachingCredentialsProvider.<init> + 190(CognitoCachingCredentialsProvider.java:190)
       at com.pontomobi.smiles.SmilesApplication.onCreate + 152(SmilesApplication.java:152)
       at android.app.Instrumentation.callApplicationOnCreate + 1014(Instrumentation.java:1014)
       at android.app.ActivityThread.handleBindApplication + 4589(ActivityThread.java:4589)
       at android.app.ActivityThread.access$1500 + 155(ActivityThread.java:155)
       at android.app.ActivityThread$H.handleMessage + 1303(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage + 102(Handler.java:102)
       at android.os.Looper.loop + 136(Looper.java:136)
       at android.app.ActivityThread.main + 5426(ActivityThread.java:5426)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke + 515(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run + 1268(ZygoteInit.java:1268)
       at com.android.internal.os.ZygoteInit.main + 1084(ZygoteInit.java:1084)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

**Comment by palpatim** [comment]
Working on a fix for this now as part of #937

**Comment by palpatim** [comment]
We released a fix for this on [2.15.1](https://github.com/aws-amplify/aws-sdk-android/releases/tag/release_v2.15.1). Please let us know if you have any questions or continue to see crashes related to retrieving encrypted data.
